pandas/core/indexes/category.py

```diff
@@ -498,6 +498,8 @@ def get_value(self, series, key):
             k = self._convert_scalar_indexer(k, kind='getitem')
             indexer = self.get_loc(k)
             return series.iloc[indexer]
+        except AttributeError:
```

Comment:
hmm, what is the attribute error you are catching?

Reply:
The above line, `return series.iloc[indexer]`, assumes the `series` argument has implemented `iloc`. The regular index version does not make this assumption. In addition, the doc for this function reads "Fast lookup of value from 1-dimensional ndarray" so this assumption should be relaxed.

Reply:
ok,, I would then put the AttributeError catching with the Key and Type error below then.?

Reply:
The key and type errors catch exceptions from `self.get_loc`, which usually happens when using indexing like `s[s>5]` leading to a key which is like a list of booleans.

The exceptions cannot use `super.get_value(series, key)` because its implementation does not work for `CategoricalIndex` when the key is a scalar value in the index (although it does work when the key is a positional index).

They also cannot all use `super.get_value(series, indexer)` because the creation of `indexer` is contingent on `self.get_loc` not failing.

Therefore, two cases are needed.

Reply:
> The above line, return series.iloc[indexer], assumes the series argument has implemented iloc. The regular index version does not make this assumption. 

ok so IIUC, ``series`` can actually be an ``Index`` or a ``Series``? if so, then just change to using ``.take()`` I think should work. Can you try and type annotate things (and update the doc-strings). 

I don't think we would need the AttributeError catching then.

Reply:
You're right. I've applied your recommended changes and added type information (the case where `series` is an `ndarray` was actually the case in the original issue)